src/controllers/like.controller.js

```diff
@@ -1,28 +1,177 @@
-import mongoose, {isValidObjectId} from "mongoose"
-import {Like} from "../models/like.model.js"
+import { Like } from "../models/like.model.js"
+import { Comment } from "../models/comment.model.js"
+import { Video } from "../models/video.model.js"
+import { Tweet } from "../models/tweet.model.js"
+import { Types, isValidObjectId } from "mongoose"
 import {ApiError} from "../utils/ApiError.js"
 import {ApiResponse} from "../utils/ApiResponse.js"
 import {asyncHandler} from "../utils/asyncHandler.js"
 
+// toggle video like
 const toggleVideoLike = asyncHandler(async (req, res) => {
-    const {videoId} = req.params
-    //TODO: toggle like on video
-})
+    const { videoId } = req.params;
+    const userId = req.user._id;
+
+    // Check if Invalid videoId
+    if (!isValidObjectId(videoId)) {
+        throw new ApiError(400, "Invalid videoId!");
+    }
+
+    // Check if the video exists
+    const video = await Video.findById(videoId);
+    if (!video) {
+        throw new ApiError(404, "Video not found!");
+    }
+
+    // Toggle like
+    const videoLike = await Like.findOne({ video: videoId });
```

Comment:
You need to check both userID & videoId

Reply:
Why, bro? The userId is not coming from the client; it comes from the auth middleware where I've already checked all the cases.